packages/lib/src/vis/shared/XAxisZoomMesh.tsx

```diff
@@ -0,0 +1,39 @@
+import { useThree } from '@react-three/fiber';
+import { Vector3 } from 'three';
+
+import type { ModifierKey } from '../models';
+import { useZoomOnWheel } from './hooks';
+
+const ZOOM_FACTOR = 0.95;
+
+interface Props {
+  disabled?: boolean;
+  modifierKey?: ModifierKey;
+}
+
+function XAxisZoomMesh(props: Props) {
+  const { disabled, modifierKey = 'Alt' } = props;
+
+  const { width, height } = useThree((state) => state.size);
+
+  const zoomVector = (sourceEvent: WheelEvent) => {
+    const factor = sourceEvent.deltaY > 0 ? ZOOM_FACTOR : 1 / ZOOM_FACTOR;
```

Comment:
Could this be computed in `useZoomOnWheel` and passed as an argument? `ZOOM_FACTOR` is the same everywhere, so there's no point to duplicate this logic for now, I reckon.

Reply:
I changed this to return booleans instead. The conversion in `zoomVector` is then done in the callback